From a101e3f0ec19ae5cf475d1f6f4c0db7756b358fc Mon Sep 17 00:00:00 2001
From: Martin Perina <mperina@redhat.com>
Date: Tue, 15 Apr 2014 13:33:10 +0200
Subject: [RHEL6.6 PATCH 2/4 v2] Add function get_pcs_cluster_nodes

Extracts relevant parts of code into function get_pcs_cluster_nodes.
This is part of code cleanup prior to introduction of fence_kdump
configuration for generic clusters.

Related: rhbz#1083938
Signed-off-by: Martin Perina <mperina@redhat.com>
---
 mkdumprd | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/mkdumprd b/mkdumprd
index 43c526b..1f81bc6 100644
--- a/mkdumprd
+++ b/mkdumprd
@@ -1523,6 +1523,13 @@ is_pcs_fence_kdump()
     [ -f "$CLUSTER_CONFIG_FILE" ] || return 1
 }
 
+# Returns list of nodes defined in Pacemaker cluster.
+get_pcs_cluster_nodes()
+{
+    echo "xpath /cluster/clusternodes/clusternode/@name" \
+        | xmllint --shell $CLUSTER_CONFIG_FILE | grep content | cut -d'=' -f2
+}
+
 if [ -z "$MNTIMAGE" -o -z "$IMAGE" ]; then
     error "Error creating temporaries.  Try again"
     cleanup_and_exit 1
@@ -1779,7 +1786,7 @@ fi
 # handle cluster node if CLUSTER_CONFIG_FILE and fence_kudmp_send exist
 if is_pcs_fence_kdump; then
     # get cluster nodes from CLUSTER_CONFIG_FILE, get interface and ip addresses
-    nodelist=`echo "xpath /cluster/clusternodes/clusternode/@name" | xmllint --shell $CLUSTER_CONFIG_FILE | grep content | cut -d'=' -f2`
+    nodelist=$(get_pcs_cluster_nodes)
     for node in ${nodelist}; do
 	addr=`getent hosts $node | cut -d' ' -f1`
 	netdev=`/sbin/ip route get to $addr 2>&1`
-- 
1.8.3.1

